app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/topsites/TopSitesAdapter.kt

```diff
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home.sessioncontrol.viewholders.topsites
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import mozilla.components.feature.top.sites.TopSite
+import org.mozilla.fenix.home.sessioncontrol.TopSiteInteractor
+
+class TopSitesAdapter(
+    private val interactor: TopSiteInteractor
+) : ListAdapter<TopSite, TopSiteItemViewHolder>(DiffCallback) {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TopSiteItemViewHolder {
+        val view = LayoutInflater.from(parent.context)
+            .inflate(TopSiteItemViewHolder.LAYOUT_ID, parent, false)
+        return TopSiteItemViewHolder(view, interactor)
+    }
+
+    override fun onBindViewHolder(holder: TopSiteItemViewHolder, position: Int) {
+        holder.bind(getItem(position))
+    }
+
+    private object DiffCallback : DiffUtil.ItemCallback<TopSite>() {
+        override fun areItemsTheSame(oldItem: TopSite, newItem: TopSite) =
+            oldItem.id == newItem.id
+
+        override fun areContentsTheSame(oldItem: TopSite, newItem: TopSite) =
+            oldItem.id == newItem.id
```

Comment:
```suggestion
            oldItem == newItem
```

Reply:
Fixed.

Reply:
Just found out we want to keep it oldItem.id == newItem.id since TopSite is not a data class and this would fail lint-lint's equality check.

Reply:
In that case, could you check all properties or leave a comment? If the title or url changes but the ID stays the same, the list won't be updated.

Reply:
Fixed.